jmx/src/main/java/org/jboss/as/jmx/PluggableMBeanServerImpl.java

```diff
@@ -1320,7 +1325,21 @@ public void addNotificationListener(ObjectName name, NotificationListener listen
                 throws InstanceNotFoundException {
             ClassLoader old = pushClassLoader(name);
             try {
-                delegate.addNotificationListener(name, listener, filter, handback);
+                NotificationListener notificationListener = new NotificationListener() {
+                    @Override
+                    public void handleNotification(Notification notification, Object handback) {
+                        ClassLoader previous = null;
+                        try {
+                            previous = Thread.currentThread().getContextClassLoader();
+                            Thread.currentThread().setContextClassLoader(listener.getClass().getClassLoader());
+                            listener.handleNotification(notification, handback);
+                        } finally {
+                            Thread.currentThread().setContextClassLoader(previous);
```

Comment:
This needs to use This needs to use WildFlySecurityManager.setContextClassLoaderPrivileged()

Reply:
@bstansberry  Changes done please check. Thanks.